src/pytest_plugins/test_filler/test_filler.py

```diff
@@ -487,19 +497,8 @@ def pytest_collection_modifyitems(config, items):
                 "blockchain_test"
             ].name.endswith("HIVE"):
                 items.remove(item)
-
-
-def pytest_runtest_setup(item):
-    """
-    A pytest hook called before setting up each test item.
-    """
-    if "yul" in item.fixturenames:
-        marker = pytest.mark.yul_test()
-        item.add_marker(marker)
-        if any([fork.name() in item.name for fork in get_development_forks()]):
-            if item.config.getoption("verbose") >= 1:
-                warnings.warn("Compiling Yul for with Shanghai, not the target fork.")
-            item.add_marker(pytest.mark.compile_yul_with("Shanghai"))
+        if "yul" in item.fixturenames:
+            item.add_marker(pytest.mark.yul_test)
 
```

Comment:
The `compile_yul_with()` functionality has been directly added to the `yul` fixture. It's more logical to have it there.